### PR TITLE
wins_per_coach, coach_sucess_percentage,  and testing

### PR DESCRIPTION
**Pull request by JimmyTrulock**
Modules developed or iterated:  created helper methods wins_per_coach and coach_suces_percentage, and added testing for them

Refactor implemented:  removed code from winningest_coach and worst_coach to helper methods

Other considerations:

### CODE REVIEW

File: lib/season_statable.rb [hunk 1]
```diff
@@ -6,39 +6,31 @@ def all_season_game_id(season)
     end.compact 
   end 
 
-  def winningest_coach(season)
+  def wins_per_coach(season)
```
Comment (Antoine-Aube) [inline lib/season_statable.rb:9]: definitely good to pull this out as it was done twice. Still a longer method, but I can't think of anything off the top of my head to make it shorter given the nested conditionals. Overall, this method is doing basically just one purpose which is good!

File: lib/season_statable.rb [hunk 2]
```diff
@@ -19,20 +19,20 @@ def wins_per_coach(season)
     coach_wins
   end
 
-  def winningest_coach(season)
-    coach_win_percentage = wins_per_coach(season).transform_values do |value| 
+  def coach_sucess_percentage(season)
```
Comment (Antoine-Aube) [inline lib/season_statable.rb:22]: yeah very nice here, this was also used a couple times, so this def drys up the code a bit more.

File: lib/season_statable.rb [hunk 3]
```diff
@@ -6,41 +6,33 @@ def all_season_game_id(season)
...
   def worst_coach(season)
-    coach_loss = @game_teams.each_with_object(Hash.new([0,0])) do |game, hash|
-      if all_season_game_id(season).include?(game.game_id) 
-        if game.result == "WIN"
-          hash[game.head_coach] = [1 + hash[game.head_coach][0], 1 + hash[game.head_coach][1]]
-        else
-          hash[game.head_coach] = [hash[game.head_coach][0], 1 + hash[game.head_coach][1]]
-        end
-      end
-    end
```
Comment (chuckrenny) [inline lib/season_statable.rb:37]: nice refactoring and breakdown of this behemoth.

**Comment by Antoine-Aube** [comment]
Nice work DrYing up these methods. We could probably go a bit further at some point with the methods that uses nested conditionals, but as noted in my inline comment, it has a pretty clear defined purpose which is what we want.

**Pull request merged**